### PR TITLE
blob/azureblob: Azure Blob example using AAD credentials

### PR DESCRIPTION
**Pull request by dadgar**
Requested by @vangent: https://github.com/google/go-cloud/issues/2410#issuecomment-511062585

### CODE REVIEW

**Comment by dadgar** [comment]
I signed it!

File: blob/azureblob/example_test.go [hunk 1]
```diff
@@ -109,3 +111,56 @@ func Example_openBucketFromURL() {
...
+		clientSecret = "456"
+		tenantID     = "789"
+
+		// The storage container to access.
+		containerName = "my-container"
+	)
+
+	// Get an Oauth2 token for the account for use with Azure Storage
+	ccc := auth.NewClientCredentialsConfig(clientID, clientSecret, tenantID)
+	ccc.Resource = azure.PublicCloud.ResourceIdentifiers.Storage
```
Comment (vangent) [inline blob/azureblob/example_test.go]: This constant doesn't appear to exist in the version we're using. Maybe just replace it with the constant string? (`https://storage.azure.com/`).

File: blob/azureblob/example_test.go [hunk 2]
```diff
@@ -109,3 +111,56 @@ func Example_openBucketFromURL() {
 	}
 	defer bucket.Close()
 }
+
+func ExampleOpenBucket_usingAADCredentials() {
+	const (
+		// Your Azure Storage Account.
+		accountName = azureblob.AccountName("my-account")
+
+		// Your Azure AAD Service Principal with access to the storage account
```
Comment (vangent) [inline blob/azureblob/example_test.go]: Nit: Please end comments with a period, here and below.

https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences.

File: blob/azureblob/example_test.go [hunk 3]
```diff
@@ -109,3 +111,56 @@ func Example_openBucketFromURL() {
...
+	// Note that we're not supplying azureblob.Options.Credential, so SignedURL
+	// won't work. To use SignedURL, you need a real credential (see the other
+	// example).
+	ctx := context.Background()
+	b, err := azureblob.OpenBucket(ctx, pipeline, accountName, containerName, new(azureblob.Options))
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer b.Close()
+
```
Comment (vangent) [inline blob/azureblob/example_test.go]: Nit: I think you can leave the `ReadAll` part off; there are plenty of other examples showing how to use a `blob.Bucket`.

File: go.mod [hunk 4]
```diff
@@ -24,10 +24,12 @@ require (
 	github.com/Azure/azure-sdk-for-go v30.0.0+incompatible
 	github.com/Azure/azure-service-bus-go v0.8.0
 	github.com/Azure/azure-storage-blob-go v0.6.0
-	github.com/Azure/go-autorest v12.0.0+incompatible
+	github.com/Azure/go-autorest/autorest v0.5.0
```
Comment (vangent) [inline go.mod]: Please revert all of the `go.mod` and `go.sum` changes, they shouldn't be needed.

Comment (dadgar) [reply go.mod]: The changes are to be able to reference `azure.PublicCloud.ResourceIdentifiers.Storage`. Will revert if you want

Comment (vangent) [reply go.mod]: Yes, please revert; we are doing a release soon and don't want to upgrade dependencies right now. Thanks!

**Comment by dadgar** [comment]
@vangent Thanks for the review. Addressed your comments, PTAL.

**Comment by vangent** [comment]
@googlebot ping

**Pull request merged**